### PR TITLE
Preparation for Open JDK 17

### PR DESCRIPTION
**Pull request by z-sztrom**
### Description of the Change

Changes necessary to compile and run under Open JDK 17

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: 
Roman Szturc roman.szturc.ext@ericsson.com

### CODE REVIEW

File: service/src/main/resources/application.properties [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 server.port=8080
 
-debug: false
+debug: true
```
Comment (jainadc9) [inline service/src/main/resources/application.properties:5]: Set debug to false

File: service/src/main/java/com/ericsson/eiffel/remrem/generate/App.java [hunk 2]
```diff
@@ -25,8 +25,8 @@
 import org.springframework.context.annotation.ComponentScan;
 import org.springframework.context.annotation.PropertySource;
 
-@SpringBootApplication
-@ComponentScan("com.ericsson.eiffel.remrem")
+@SpringBootApplication(scanBasePackages = {"com.ericsson.eiffel.remrem", "com.ericsson.eiffel.remrem.semantics"})
+@ComponentScan({"com.ericsson.eiffel.remrem", "com.ericsson.eiffel.remrem.semantics"})
```
Comment (dhruvin-kalavadia) [inline service/src/main/java/com/ericsson/eiffel/remrem/generate/App.java:29]: The @SpringBootApplication annotation already includes the @ComponentScan annotation. So I believe there is no specific need to add the @ComponentScan annotation separately as the same packages are mentioned with scanBasePackages for the @SpringBootApplication annotation.

File: service/pom.xml [hunk 3]
```diff
@@ -131,13 +150,20 @@
...
+
+                </configuration>
             </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-failsafe-plugin</artifactId>
                 <!-- Integration tests are failing with 2.21.0 version of maven-failsafe-plugin 
                     so using 2.20.1 version -->
                 <version>2.22.0</version>
+<!--                <version>3.0.0-M7</version>-->
```
Comment (dhruvin-kalavadia) [inline service/pom.xml:166]: Please remove this commented version.

**Review by dhruvin-kalavadia — COMMENTED** [review]
Please change the title to Upgrading to OpenJDK17

File: service/src/main/java/com/ericsson/eiffel/remrem/generate/App.java (hunk 2 above)
Comment (z-sztrom) [reply service/src/main/java/com/ericsson/eiffel/remrem/generate/App.java:29]: You're right. I was unable to get it working unless both of them were presented. It works fine now with @SpringBootApplication only. Removing @ComponentScan.

File: service/pom.xml (hunk 3 above)
Comment (z-sztrom) [reply service/pom.xml:166]: Removed.

**Review by dhruvin-kalavadia — APPROVED** [review]

**Review by jainadc9 — APPROVED** [review]

**Pull request merged**